coil-base/src/main/java/coil/fetch/HttpFetcher.kt

```diff
@@ -15,17 +17,38 @@ import okhttp3.HttpUrl
 import okhttp3.Request
 import okhttp3.ResponseBody
 
-internal class HttpUrlFetcher(private val callFactory: Call.Factory) : Fetcher<HttpUrl> {
+internal class HttpUriFetcher(callFactory: Call.Factory) : HttpFetcher<Uri>(callFactory) {
```

Comment:
Restoring the optimization to call `HttpUrl.get` as part of `Fetcher.fetch` instead of part of `Mapper.map` as `Fetcher.fetch` is always called on a background thread.